Sources/NIOUDPEchoClient/main.swift

```diff
@@ -0,0 +1,147 @@
+ //===----------------------------------------------------------------------===//
+ //
+ // This source file is part of the SwiftNIO open source project
+ //
+ // Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+ // Licensed under Apache License v2.0
+ //
+ // See LICENSE.txt for license information
+ // See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+ //
+ // SPDX-License-Identifier: Apache-2.0
+ //
+ //===----------------------------------------------------------------------===//
+ import NIO
+ 
+ print("Please enter line to send to the server")
+ let line = readLine(strippingNewline: true)!
+ 
+ private final class EchoHandler: ChannelInboundHandler {
+    public typealias InboundIn = AddressedEnvelope<ByteBuffer>
+    public typealias OutboundOut = AddressedEnvelope<ByteBuffer>
+    private var numBytes = 0
+    
+    private let remoteAddressInitializer: () throws -> SocketAddress
+    
+    init(remoteAddressInitializer: @escaping () throws -> SocketAddress) {
+        self.remoteAddressInitializer = remoteAddressInitializer
+    }
+
+    public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let envelope = self.unwrapInboundIn(data)
+        var byteBuffer = envelope.data
+
+        numBytes -= byteBuffer.readableBytes
+
+        assert(numBytes >= 0)
```

Comment:
I'm wondering if we should assert here or not, because that makes this crash if the server is sending us too much data...

Reply:
I agree. I don't think it adds anything. I just added it to match the TCP example. Will remove.